antigen_bot/bot.py

```diff
@@ -112,15 +115,17 @@ async def on_message(self, msg: Message) -> None:
             if msg.type() == MessageType.MESSAGE_TYPE_URL:
                 urlfile = await msg.to_url_link()
                 random.shuffle(self.user_send_quns[talker.contact_id])
-                for room in self.user_send_quns[talker.contact_id]:
+                for room_id in self.user_send_quns[talker.contact_id]:
+                    room = await xiaoyan.Room.find(room_id)
                     try:
                         await room.say(urlfile)
                     except Exception as e:
                         print(e)
 
             if msg.type() == MessageType.MESSAGE_TYPE_TEXT:
                 random.shuffle(self.user_send_quns[talker.contact_id])
-                for room in self.user_send_quns[talker.contact_id]:
+                for room_id in self.user_send_quns[talker.contact_id]:
```

Comment:
I suggest:

```python
self.user_send_quns.get(talker.contact_id, [])
```

to avoid the latent bugs which the `contact_id` is not in `user_send_quns`.